src/Lykke.Service.GoogleAnalyticsWrapper.Services/ShutdownManager.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Common;
+using Common.Log;
+using Lykke.Service.GoogleAnalyticsWrapper.Core.Services;
+
+namespace Lykke.Service.GoogleAnalyticsWrapper.Services
+{
+    // NOTE: Sometimes, shutdown process should be expressed explicitly. 
+    // If this is your case, use this class to manage shutdown.
+    // For example, sometimes some state should be saved only after all incoming message processing and 
+    // all periodical handler was stopped, and so on.
+    
+    public class ShutdownManager : IShutdownManager
+    {
+        private readonly ILog _log;
+        private readonly List<IStopable> _items = new List<IStopable>();
+
+        public ShutdownManager(ILog log)
```

Comment:
Autofac can by self fill a collection of IStopable.
public ShutdownManager(ILog log, IEnumerable<IStopable>) should work